numpyro/optim.py

```diff
@@ -346,16 +346,17 @@ def optax_to_numpyro(transformation) -> _NumPyroOptim:
         Optax optimizer.
     """
     import optax
+    transformation = optax.with_extra_args_support(transformation)
 
     def init_fn(params: _Params) -> tuple[_Params, Any]:
         opt_state = transformation.init(params)
         return params, opt_state
 
     def update_fn(
-        step: ArrayLike, grads: ArrayLike, state: tuple[_Params, Any]
+        step: ArrayLike, grads: ArrayLike, state: tuple[_Params, Any], value
     ) -> tuple[_Params, Any]:
         params, opt_state = state
-        updates, opt_state = transformation.update(grads, opt_state, params)
+        updates, opt_state = transformation.update(grads, opt_state, params, value=value)
```

Comment:
maybe: optax.with_extra_args_support(transformation).update(...)

Reply:
I'm not sure. it would be surprised to me if the optimizer is stateful. but either way looks good to me.

Reply:
Yes, the transformation objects are stateless.